src/app/shared/services/history.service.ts

```diff
@@ -13,7 +15,33 @@ export class HistoryService {
   constructor(private afs: AngularFirestore) {
   }
 
-  getHistory(top: number): Observable<HistoryModel[]> {
+  private static convertUserHistory(history: HistoryModel): GenericHistoryModel {
+    const tmp = new GenericHistoryModel();
+    tmp.created = history.created;
+    tmp.lastUpdated = history.lastUpdated;
+    tmp.winnerName = history.player1Name;
+    tmp.winnerOldRating = history.player1OldRating;
+    tmp.winnerNewRating = history.player1NewRating;
+    tmp.loserName = history.player2Name;
+    tmp.loserOldRating = history.player2OldRating;
+    tmp.loserNewRating = history.player2NewRating;
+    return tmp;
+  }
+
+  private static convertTeamHistory(history: TeamHistoryModel): GenericHistoryModel {
```

Comment:
I find the regular object creation to be easier to read than editing the object for each property. Also, this is more in tune of how the project's code is structured (https://github.com/palexcast/foos-rating/blob/df518c47d6e60b18bc5f07bc209e569b15c18cf3/src/app/home/register-winner/register1vs1-match/register1vs1-match.component.ts#L64-L71).

```javascript
return {
  created: history.created,
  lastUpdated: history.lastUpdated,
  winnerName: history.team1Name,
  winnerOldRating: history.team1OldRating,
  winnerNewRating: history.team1NewRating,
  loserName: history.team2Name,
  loserOldRating: history.team2OldRating,
  loserNewRating: history.team2NewRating
}
```